src/transaction.js

```diff
@@ -222,4 +226,57 @@ export class Transaction extends TransactionBase {
 
     return envelope;
   }
+
+  /**
+   * Calculate the claimable balance ID for an operation within the transaction.
+   *
+   * This helper throws under the following circumstances:
+   *    - invalid `opIndex` value
+   *    - the operation at the `opIndex` is not a CreateClaimableBalance op
+   *    - the source account doesn't have a sequence number
+   *    - general XDR un/marshalling failures
+   *
+   * @param   {integer}  opIndex   the index of the CreateClaimableBalance op
+   * @returns {string}   a hex string representing the claimable balance ID
+   *
+   * @see https://github.com/stellar/go/blob/master/txnbuild/transaction.go#L392
+   */
+  getClaimableBalanceId(opIndex) {
+    // Validate and then extract the operation from the transaction.
+    if (
+      !Number.isInteger(opIndex) ||
+      opIndex < 0 ||
+      opIndex >= this.operations.length
+    ) {
+      throw new RangeError('invalid operation index');
+    }
+
+    let op = this.operations[opIndex];
+    try {
+      op = Operation.createClaimableBalance(op);
+    } catch (err) {
+      throw new TypeError(
+        `expected createClaimableBalance, got ${op.type}: ${err}`
+      );
+    }
+
+    // Use the operation's source account or the transaction's source if not. In
+    // either case, use the unmuxed version.
+    let account = decodeAddressToMuxedAccount(this.source, true);
+    if (account.switch() === xdr.CryptoKeyType.keyTypeMuxedEd25519()) {
+      account = account.med25519();
+    }
```

Comment:
👀  I don't see you trying to use the op source account here, it seems you always get the transaction source. Is this comment accurate? Shouldn't it be referring to the muxed account G... address?

Reply:
Good catch, the comment is outdated! We should always use the G-address.